packages/wonder-blocks-dropdown/src/components/multi-select.tsx

```diff
@@ -449,79 +417,50 @@ export default class MultiSelect extends React.Component<Props, State> {
 
         return [
             ...lastSelectedItems,
-            ...restOfTheChildren.map(this.mapOptionItemToDropdownItem),
+            ...restOfTheChildren.map(mapOptionItemToDropdownItem),
         ];
-    }
-
-    mapOptionItemToDropdownItem: (option: OptionItemComponent) => DropdownItem =
-        (option: OptionItemComponent): DropdownItem => {
-            const {selectedValues} = this.props;
-            const {disabled, value} = option.props;
-            return {
-                component: option,
-                focusable: !disabled,
-                populatedProps: {
-                    onToggle: this.handleToggle,
-                    selected: selectedValues.includes(value),
-                    variant: "checkbox",
-                },
-            };
+    };
+
+    const mapOptionItemToDropdownItem = (
+        option: OptionItemComponent,
+    ): DropdownItem => {
+        const {disabled, value} = option.props;
+        return {
+            component: option,
+            focusable: !disabled,
+            populatedProps: {
+                onToggle: handleToggle,
+                selected: selectedValues.includes(value),
+                variant: "checkbox",
+            },
         };
+    };
 
-    handleOpenerRef: (node?: any) => void = (node: any) => {
+    const handleOpenerRef = (node?: any) => {
         const openerElement = ReactDOM.findDOMNode(node) as HTMLElement;
-        this.setState({openerElement});
+        setOpenerElement(openerElement);
     };
 
-    handleSearchTextChanged: (searchText: string) => void = (
-        searchText: string,
-    ) => {
-        this.setState({searchText});
+    const handleSearchTextChanged = (searchText: string) => {
+        setSearchText(searchText);
     };
 
-    handleClick: (e: React.SyntheticEvent) => void = (
-        e: React.SyntheticEvent,
-    ) => {
-        this.handleOpenChanged(!this.state.open);
+    const handleClick = (e: React.SyntheticEvent) => {
+        handleOpenChanged(!open);
     };
 
-    renderOpener(
+    const renderOpener = (
         allChildren: React.ReactElement<
             React.ComponentProps<typeof OptionItem>
         >[],
         isDisabled: boolean,
         dropdownId: string,
     ):
         | React.ReactElement<React.ComponentProps<typeof DropdownOpener>>
-        | React.ReactElement<React.ComponentProps<typeof SelectOpener>> {
-        const {
-            id,
-            light,
-            opener,
-            testId,
-            // the following props are being included here to avoid
```

Comment:
Since we deconstruct the props at the start of the function, we don't need to again in the renderOpener function. 
https://github.com/Khan/wonder-blocks/blob/32b4f84272bf6c7b98e904fb9a87260f6ae467fd/packages/wonder-blocks-dropdown/src/components/multi-select.tsx#L194-L218

The deconstruction previously is almost the same. The FC will also extract `disabled`, `error`, `children`, and `dropdownId` so they won't be included in `sharedProps` when it is spread. The only one that matters to the SelectOpener is `error`, which is why we also explicitly set the `error` prop: https://github.com/Khan/wonder-blocks/pull/2370/files#diff-f7d0f4de4f6447c1f878dd6c51e6646adae276658cf9d7f29fcea7f630b17430R484

(changes for the `disabled` doesn't matter since we were already explicitly setting it https://github.com/Khan/wonder-blocks/pull/2370/files#diff-f7d0f4de4f6447c1f878dd6c51e6646adae276658cf9d7f29fcea7f630b17430R485)